api/views.py

```diff
@@ -0,0 +1,258 @@
+from rest_framework.views import APIView
+from django.contrib.auth import authenticate, login
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework.decorators import api_view, renderer_classes, authentication_classes, permission_classes, throttle_classes, parser_classes
+from rest_framework.response import Response
+from rest_framework.authtoken.models import Token
+
+from posts.views import post_content_is_valid, parse_content, save_hashtags, post_image_is_valid
+from circles.models import Circle
+from posts.models import Post
+
+post_required_values = lambda user_circles_json: {
+    'content': {
+        'Description': 'Can contain mentions and hashtags.',
+        'Restrictions': 'max length 256 unicode characters and min length 1 character',
+    },
+    'circle': {
+        'Description': 'Determines who can see the post.',
+        'Restrictions': 'Select exactly one, if set to 1 then it is going to be in the circle \'Me\'',
+        'Values': user_circles_json,
+    },
+    'image': {
+        'Description': 'An image file can be uploaded, unfortunately not via JSON',
+        'Restrictions': 'If none given, please write None',
+    },
+}
+
+
+class RootView(APIView):
+    """
+    Generic View of the root view, which contains the use documentation of the API, it gives it as a Browsable API or JSON.
+
+    This can be requested from anyone and renders the requests from JSON and Forms.
+    """
+    renderer_classes = (renderers.BrowsableAPIRenderer, renderers.JSONRenderer,)
+    permissions_classes = permissions.AllowAny
+    parser_classes = parsers.JSONParser, parsers.FormParser
+
+    def get(self, request, *args, **kwargs):
+        # TODO: add about page
+        return Response({
+            'API options': {
+                'login': {
+                    'URL': '/api/login/',
+                    'Methods Allowed': 'GET, POST',
+                    'Required data': {
+                        'email': 'The registered email for the user\'s account',
+                        'password': 'The password for the account',
+                    },
+                },
+                'create': {
+                    'url': '/api/create/',
+                    'Methods Allowed': 'POST',
+                    'Required data': {
+                        'content': 'Maximum length of 256 unicode characters. It is the post content and it can contain text, mentions, and hashtags',
+                        'image': 'An uploaded image or None',
+                        'circles': 'The int value of a circle. Default values are:\n 0 -> Me circle \n 1 -> Public circle',
+                    },
+                },
+                'Documentation': 'Please read our Manual here: https://github.com/7Pros/circuit/blob/develop/MANUAL.md',
+                'Contact': 'Please open an issue on our issue tracker: https://github.com/7Pros/circuit/issues/new',
+            },
+        })
+
+
+@api_view(['GET', 'POST'])
```

Comment:
This says that I can make a GET request to `/api/login`. But when I do I get the following error.

![download](https://cloud.githubusercontent.com/assets/872251/8566145/44af484c-255c-11e5-8668-3cfb9bf8172c.png)